ts/Extensions/Annotations/Controllables/ControllableEllipse.ts

```diff
@@ -0,0 +1,297 @@
+    public transform = ControllableMixin.transform;
+    public translatePoint = ControllableMixin.translatePoint;
+    public transformPoint = ControllableMixin.transformPoint;
+
+    public update(
+        this: ControllableEllipse,
+        newOptions: Highcharts.AnnotationControllableOptionsObject
+    ): void {
+        const options = merge(true, this.options, newOptions);
+        this.options = options;
+        this.redraw();
+    }
+
+    /**
+     * @type 'ellipse'
+     */
+    public type = 'ellipse';
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(annotation: Annotation, options: EllipseShapeOptions, index: number): void {
+        if (defined(options.yAxis)) {
+            (options.points as MockPointOptions[]).forEach((point): void => {
+                point.yAxis = options.yAxis;
+            });
+        }
+
+        if (defined(options.xAxis)) {
+            (options.points as MockPointOptions[]).forEach((point): void => {
+                point.xAxis = options.xAxis;
+            });
+        }
+
+        ControllableMixin.init.call(this, annotation, options, index);
+    }
+
+    /**
+     *
+     * Render the element
+     * @param parent parent SVG element.
+     */
+    public render(parent: SVGElement): void {
+        this.graphic = this.annotation.chart.renderer.createElement('ellipse')
+            .attr(this.attrsFromOptions(this.options))
+            .add(parent);
+
+        ControllableMixin.render.call(this);
+    }
+
+    /**
+     * Translate the points.
+     * Mostly used to handle dragging of the ellipse.
+     */
+
+    public translate(this: ControllableEllipse, dx: number, dy: number): void {
+        ControllableMixin.translateShape.call(this, dx, dy, true);
+    }
+
+    /**
+     * Get the distance from the line to the point.
+     * @param point1 first point which is on the line
+     * @param point2 second point
+     * @param x0 point's x value from which you want to calculate the distance from
+     * @param y0 point's y value from which you want to calculate the distance from
+     */
+    public getDistanceFromLine(
+        point1: ReferencePointsOptions,
+        point2: ReferencePointsOptions,
+        x0: number,
+        y0: number
+    ): number {
+        return Math.abs(
+            (point2.y - point1.y) * x0 - (point2.x - point1.x) * y0 +
+            point2.x * point1.y - point2.y * point1.x
+        ) / Math.sqrt(
+            (point2.y - point1.y) * (point2.y - point1.y) +
+                (point2.x - point1.x) * (point2.x - point1.x)
+        );
+    }
+
+    /**
+     * The fuction calculates the svg attributes of the ellipse, and returns all
+     * parameters neccessary to draw the ellipse.
+     * @param position absolute position of the first point in points array
+     * @param position2 absolute position of the second point in points array
+     */
+    public getAttrs(position: BBoxObject, position2: BBoxObject): EllipseShapeSVGOptions {
```

Comment:
Discussion: I'm wondering whether `position` and `position2` are good variables names

Reply:
any suggestions?
@pawelfus @karolkolodziej @raf18seb

Reply:
If I were to change it, I would go with `firstPointPostion` and `secondPointPosition`.